src/build/__main__.py

```diff
@@ -126,14 +128,84 @@ def build_package(srcdir, outdir, distributions, config_settings=None, isolation
+    :param isolation: Isolate the build in a separate environment
+    :param skip_dependency_check: Do not perform the dependency check
+    """
+    builder = ProjectBuilder(srcdir)
+    for distribution in distributions:
+        _build(isolation, builder, outdir, distribution, config_settings, skip_dependency_check)
+
+
+def build_package_via_sdist(srcdir, outdir, distributions, config_settings=None, isolation=True, skip_dependency_check=False):
+    # type: (str, str, List[str], Optional[ConfigSettingsType], bool, bool) -> None
+    """
+    Build a sdist and then the specified distributions from it.
+
+    :param srcdir: Source directory
+    :param outdir: Output directory
+    :param distribution: Distribution to build (only wheel)
+    :param config_settings: Configuration settings to be passed to the backend
+    :param isolation: Isolate the build in a separate environment
+    :param skip_dependency_check: Do not perform the dependency check
+    """
+    if 'sdist' in distributions:
+        raise ValueError('Only binary distributions are allowed but sdist was specified')
+
+    builder = ProjectBuilder(srcdir)
+    sdist = _build(isolation, builder, outdir, 'sdist', config_settings, skip_dependency_check)
+
+    # extract sdist
+    sdist_name = os.path.basename(sdist)
+    sdist_out = tempfile.mkdtemp(dir=outdir, prefix='build-via-sdist-')
+    with tarfile.open(sdist) as t:
+        t.extractall(sdist_out)
+        builder = ProjectBuilder(os.path.join(sdist_out, sdist_name[: -len('.tar.gz')]))
+        for distribution in distributions:
+            _build(isolation, builder, outdir, distribution, config_settings, skip_dependency_check)
+
+    # remove sdist source if there was no exception
+    shutil.rmtree(sdist_out, ignore_errors=True)
+
+
 def main_parser():  # type: () -> argparse.ArgumentParser
     """
     Construct the main parser.
     """
     # mypy does not recognize module.__path__
     # https://github.com/python/mypy/issues/1422
     paths = build.__path__  # type: Iterable[Optional[str]]  # type: ignore
-    parser = argparse.ArgumentParser()
+    parser = argparse.ArgumentParser(
+        description=_indent(  # textwrap.indent
+            textwrap.dedent(
+                '''
+                A simple, correct PEP 517 package builder.
+
+                This CLI tool can be used to generate Python package distributions
+                (sdists and wheels) as defined by PEP 517.
+
+                By default, it will build a source distribution (sdist) for the
+                project and then build the binary distributions (wheels).
+                This is recommended as it will ensure your source distribution (sdist)
+                can be used to build wheels.
+
+                If any distribution flag is passed, -s/--sdist or -w/--wheel, only
+                those distributions will be built, and they will be built directly
+                from the specified source.
```

Comment:
I think this a little hazy and repetitive, let's cut it down. Maybe:

```suggestion
                By default, a source distribution (sdist) is built from `srcdir` and
                a binary distribution (wheel) is built from the sdist.
                This is recommended as it will ensure the sdist can be used
                to build wheels.
                Pass `-s/--sdist` or `-w/--wheel` to build a specific
                distribution. If you pass `-w/--wheel`, the wheel
                will be built from `srcdir` and not from an sdist, even if you
                combine it with `-s/--sdist`.
```

Reply:
I have rewritten as follows.

```
By default, a source distribution (sdist) is built from {srcdir}
and a binary distribution (wheel) is built from the sdist.
This is recommended as it will ensure the sdist can be used
to build wheels.

Pass -s/--sdist and/or -w/--wheel to build a specific distribution.
If you do this, the default behavior will be disabled, and all
artifacts will be built from {srcdir} (even if you combine
-w/--wheel with -s/--sdist, the wheel will be built from {srcdir}).
```

I used the `{srcdir}` notation to match what we already had.

Reply:
Sounds good, thanks.